funcx_endpoint/funcx_endpoint/endpoint/endpoint.py

```diff
@@ -314,68 +310,51 @@ def start_endpoint(
-            reg_info = None
-
-
-def handle_start_registration(funcx_client, name, endpoint_uuid, endpoint_dir, retry=False):
+        reg_info = handle_start_registration(funcx_client, name, endpoint_uuid, endpoint_dir, previously_registered)
+
+        # Configure the parameters for the interchange
+        optionals = {}
+        optionals['client_address'] = reg_info['public_ip']
+        optionals['client_ports'] = reg_info['tasks_port'], reg_info['results_port'], reg_info['commands_port'],
+        if 'endpoint_address' in State.FUNCX_CONFIG:
+            optionals['interchange_address'] = State.FUNCX_CONFIG['endpoint_address']
+
+        optionals['logdir'] = endpoint_dir
+
+        if State.DEBUG:
+            optionals['logging_level'] = logging.DEBUG
+
+        ic = EndpointInterchange(endpoint_config.config,
+                                 endpoint_id=endpoint_uuid,
+                                 keys_dir=keys_dir,
+                                 **optionals)
+        ic.start()
+        ic.stop()
+
+        logger.critical("Interchange terminated.")
+
+
+def handle_start_registration(funcx_client, name, endpoint_uuid, endpoint_dir, previously_registered):
     # Register the endpoint
     logger.info("Registering endpoint")
     if State.FUNCX_CONFIG.get('broker_test', False) is True:
         logger.warning("**************** BROKER State.DEBUG MODE *******************")
         reg_info = register_with_hub(endpoint_uuid,
                                      endpoint_dir,
                                      State.FUNCX_CONFIG['broker_address'],
-                                     State.FUNCX_CONFIG['redis_host'])
-    elif retry:
-
-        # this will retry the registration process every 5s until it succeeds
-        while True:
-            try:
-                reg_info = register_endpoint(funcx_client, name, endpoint_uuid, endpoint_dir)
-                break
-            except Exception as e:
-                logger.critical(e)
-                logger.warning("Retrying registration in 5s")
-                time.sleep(5)
-                continue
+                                     client_public_key=client_public_key,
+                                     redis_host=State.FUNCX_CONFIG['redis_host'])
     else:
-        reg_info = register_endpoint(funcx_client, name, endpoint_uuid, endpoint_dir)
+        reg_info = register_endpoint(funcx_client, name, endpoint_uuid, endpoint_dir, previously_registered)
 
     logger.info("Endpoint registered with UUID: {}".format(reg_info['endpoint_id']))
 
     return reg_info
 
 
 # Avoid a race condition when starting the endpoint alongside the web service
-def register_endpoint(funcx_client, endpoint_name, endpoint_uuid, endpoint_dir):
+@retry(delay=10, max_delay=300, backoff=1.2, logger=logging.getLogger('funcx'))
+def register_endpoint(funcx_client, endpoint_name, endpoint_uuid, endpoint_dir, previously_registered):
```

Comment:
Please check the docstring to make sure it is consistent with the fn definition.

Reply:
The function definition has been rolled back to the original one. So the docstring is okay now.